ibazel/main.go

```diff
@@ -94,27 +94,30 @@ func main() {
 		return
 	}
 
+	// Create a timer objet to use for debouncing. When the timer is created, do
+	// not start the timer but instead start the timer when an event that builds
+	// should be run.  This has the added benefit of kicking things off by
+	// sending an event to make the first run happen.
+	debouncer := time.NewTimer(debounceDuration)
+
 	// Listen to the events and trigger action based on the response code.
-	go func() {
-		for {
-			select {
-			case _ = <-watcher.Events:
-				fmt.Printf("Files changed rebuilding...\n")
-				commandToRun(b, target)
-			case err := <-watcher.Errors:
-				fmt.Println("Error:", err)
-			}
+	for {
+		select {
+		case _ = <-watcher.Events:
+			fmt.Printf("Files changed...\n")
+			debouncer.Reset(debounceDuration)
```

Comment:
See discussion in https://golang.org/pkg/time/#Timer.Reset re: Stop and Reset.  I'm worried that you'll potentially have a full debouncer.C when calling Reset, in certain conditions where the watcher fires right when debouncer's time runs out, and you hit the first case instead of the second.